Jenkinsfile

```diff
@@ -1,44 +1,50 @@
 pipeline {
```

Comment:
Before merging, since we want to maintain the old functionality, can this be renamed to `Jenkinsfile.new` so we can keep the current framework running? We can then slowly move functionality over to the current Jenkinsfile as we validate tests and resolve issues blocking us from using this new framework. I'm open to discussion on this, so those who are reviewing as well, please comment / provide feedback.